src/views/List.tsx

```diff
@@ -1,21 +1,35 @@
-import React from 'react';
-import { ListItem as ListItemComponent } from '../components/ListItem';
-import { ListItem } from '../api';
+import { useState } from "react";
+import { ListItem as ListItemComponent } from "../components/ListItem";
+import { FilterList as FilterListComponent } from "../components/FilterList";
+import { ListItem } from "../api";
 
 interface Props {
 	data: ListItem[];
 }
 
 export function List({ data }: Props) {
-	const hasItem = data.length !== 0;
+	const [searchTerm, setSearchTerm] = useState<string>("");
+
+	const filteredData = data.filter((item) =>
+		item.name.toLowerCase().includes(searchTerm.toLowerCase()),
+	);
+
+	const hasItem = filteredData.length !== 0;
```

Comment:
This I think is the somewhat dangerous part of using variables like this: they can be misleading. This is _really_ `hasFilteredItems`, but even that is hard to mentally grasp onto when we also want to conditionally render `FilterListComponent` if there are any items (regardless of filter). Imo I think it would be easier to understand what this component is doing if we remove the conditional render on line 31 and let `filteredData.map` return an empty array when there's nothing in `filteredData`; it will have the same effect without the indirection introduced by `hasItem` and make conditionally rendered `FilterListComponent` under separate but similar conditions easier to understand what makes them different from each other

Reply:
This is a really good point, and ties in with your comment about renaming the prop on the fly, and @RossaMania comment on updating the name of the `filterdData` as well. I think all those changes make the flow of the code easier to read!